### ISSUE TITLE
Stop using magic to make code unreadable

### ISSUE THREAD

**Issue by nickvergessen** [SergeyMosin/Appointments#182]
https://github.com/SergeyMosin/Appointments/blob/47067a400bba35058030d73da94f4bd04e8b4393/lib/Backend/TalkIntegration.php#L105-L108

What is this trying to do, just write it in your code and stop obfuscating it

**Comment by SergeyMosin** [comment]
This contributor token verification code.

**Comment by SergeyMosin** [comment]
Only contributors can automatically add passwords and lobby

**Comment by nickvergessen** [comment]
But why:
```php
 $n="getUs"."erValue";$hd='he'."xdec"; 
 $c=$this->config->$n($userId, $this->appName, 'c'."nk"); 
 $sss="su".'bstr'; 
 if(!empty($roomToken) && $c!=='' && (($hd($sss($c,0,0b100))>>14)& 1)===(($hd($sss($c,4,4))>>  6) & 1) && isset($c[5])){ 
```
instead of:
```php
 $c=$this->config->getUserValue($userId, $this->appName, 'cnk');
 if(!empty($roomToken) && $c!=='' && ((hexdec(substr($c,0,0b100))>>14)& 1)===((hexdec(substr($c,4,4))>>  6) & 1) && isset($c[5])){ 
```

**Comment by SergeyMosin** [comment]
obfuscating

**Comment by nickvergessen** [comment]
Yeah don't do that.

**Comment by nolens** [comment]
@nickvergessen 
The goal is to give privileges to those who contribute financially or with code....

**Comment by nickvergessen** [comment]
> The goal is to give privileges to those who contribute financially or with code....

So shall I do that too? I rename my internal methods in Talk on every patch release unless you contribute or finance? Sorry but this is not my understanding of open source at least. If your app works people give back, but making it hard for people to help and Nextcloud people to review and give hints, that's a wrong reason and strategy.

**Comment by nickvergessen** [comment]
I mean, I could warn you about things like breaking Talk integration when I could read the code:  https://github.com/SergeyMosin/Appointments/issues/179 (Fix not in any PR: https://github.com/SergeyMosin/Appointments/commit/47067a400bba35058030d73da94f4bd04e8b4393 )

But I'm not going to invest a minute into decrypting code or check if an app works when it looks like this.

**Comment by nolens** [comment]
Sorry, I probably should not have commented on this. I am really _only_ a user who did some initial funding of the plugin via a funding platform.... I am sure @SergeyMosin will also comment on this, Sergey is a guy with good intentions and usually pretty responsive.

**Comment by LittleDuke** [comment]
> Sorry, I probably should not have commented on this. I am really _only_ a user who did some initial funding of the plugin via a funding platform.... I am sure @SergeyMosin will also comment on this, Sergey is a guy with good intentions and usually pretty responsive.

I too REGULARLY donate to this project -- I find it has high value which is why at the moment the latest release broke for me and I'm unable to get people onto my calendar which I know @SergeyMosin is responsive and I will certainly reward his time and attention again if he gets us back online today -- everyday that goes by means lost revenue for us and at some point we'll be forced to go license Calendly.

Making the code harder to maintain is NOT in anyone's best interests IMHO especially if it breaks things.  

That's not the kind of drag or friction a good app needs..

-dvd

**Comment by SergeyMosin** [comment]
> I rename my internal methods in Talk on every patch release unless you contribute or finance?
> I agree it's a shitty thing to do -- lots of modules have paid keys that use digital signatures without having to resort to trying to obfuscate the code

**Only code related to CHECKING THE KEYS is obfuscated**, all other code is NOT, plus it is not that hard to figure out what's going for anyone with adequate knowledge of PHP, see here for example https://github.com/SergeyMosin/Appointments/issues/182#issuecomment-787918024

> Sorry but this is not my understanding of open source at least.

Once again **ONLY KEY VERIFICATION CODE IS OBFUSCATED**, also please see the explanation here https://github.com/SergeyMosin/Appointments/issues/79#issuecomment-687659269 , point 4 says `Contact me if any of the above methods do not work for you`, and trust me more often than not I DO give away free keys. I addition there is not a word about restrictions to what you can do with the key, If for some reason you decide share it with everyone in China, there is no body but yourself to stop you from doing so.

**Comment by jospoortvliet** [comment]
Hey @SergeyMosin what I would recommend is you just skip the obfuscation - as you say, somebody who knows PHP can take it out if they want. Just ship the app in the app store with the key check. Most people won't go and change it - and those who want to change it can do it, obfuscation or not. Just rely on the lazyness of most people. That is what apps do that charge money on the Google app store but are for free in F-droid too, it works perfectly fine.

I think that the cost of irritating some open source people (like @nickvergessen ) is bigger than that it helps differentiate the benefits. I would even go further and just put a comment there, with something like:

> this code checks for a key that unlocks a few features that we give to the awesome contributors to our project. If you've send a patch, helped translate, or contributed in any way, please just ask me for a code! If you can't contribute code and can't afford a little donation, just ask me and I will happily give you a code for free. I never say no to a request and you don't need to explain yourself!

-> I know some ppl (like Sam Harris) that do that. Just say: if you can't afford to contribute, I give you a free code, just ask. This friendlyness is often rewarded. People feel embarrassed to ask for something for free if they can afford to contribute just fine and they rather pay a few euros. I would even think that this gets you MORE contributions than obfuscated code. With this, you give people guilt and enjoyment of contributing. With obfuscated code, you give them a challenge which they might enjoy to solve, without any guilt, rather than contribute.

Just some thoughts ;-)

**Comment by Bellusterra** [comment]
> Once again ONLY KEY VERIFICATION CODE IS OBFUSCATED, also please see the explanation here #79 (comment) , point 4 says Contact me if any of the above methods do not work for you, and trust me more often than not I DO give away free keys. I addition there is not a word about restrictions to what you can do with the key, If for some reason you decide share it with everyone in China, there is no body but yourself to stop you from doing so.

Contributor key:
843f1-d2b29-e3027-f6cd1

All jokes aside, @jospoortvliet @nickvergessen may I suggest, if not already implemented, some form of a financial reward system for App Developers.? The Applications and of course those making them make up a big part of what Nextcloud is, an amazing project filled with amazing people. I would think at the opportunity cost of hiring on another full time employee at Nextcloud something could be implemented maybe based on App use or rating or whatever to support Developers that add to the Nextcloud ecosystem. Even having a public contribution page that gets paid out to the App Developers would be a good idea I think. I believe a lot of people would like/want to contribute if there was an easy way to do so.

@SergeyMosin I want to say thank you for making Nextcloud Appointments, people like you are what make Nextcloud and open source so great. I for one was happy to contribute regardless of any extra features or otherwise.

**Comment by spaetz** [comment]
I just donated a little, thanks! I think there is a difference between obfuscating code and having 2 lines that make it a little harder to get premium features by freeriding. I would be with @nickvergessen if the obfuscation would extend other code areas, but given that this is a slight obfuscation in 2 Loc that have no other functionality, is really nothing to complain about imho.

@Bellusterra, I did not get the joke 🤷. You just spoiled the low hurdle to actually nudge people into contriburing financially.

@SergeyMosin , I was looking for a donation link in your readme.md and did not find any. It took me a while until I found the github donation link feature, perhaps place that **a little** more prominently?

Anyway, I guess this is a case of "move on, nothing to see here".
